### PR TITLE
CMusicInfoTag: copy some missing information in SetArtist/SetAlbum/SetSong()

### PR DESCRIPTION
**Pull request by Montellese**
While investigating http://trac.kodi.tv/ticket/16232 I noticed that quite a few details available in `CArtist`, `CAlbum` and `CSong` aren't copied to `CMusicInfoTag` in `SetArtist()`, `SetAlbum()` and `SetSong()` so I added all that I think are safe (wasn't sure about copying the path).

### CODE REVIEW

File: xbmc/music/tags/MusicInfoTag.cpp [hunk 1]
```diff
@@ -581,11 +581,13 @@ void CMusicInfoTag::SetArtist(const CArtist& artist)
 {
   SetArtist(artist.strArtist);
   SetAlbumArtist(artist.strArtist);
+  SetMusicBrainzArtistID({ artist.strMusicBrainzArtistID });
   SetGenre(artist.genre);
-  m_dateAdded = artist.dateAdded;
-  m_iDbId = artist.idArtist;
-  m_type = MediaTypeArtist;
-  m_bLoaded = true;
+  SetMood(StringUtils::Join(artist.moods, " / "));
```
Comment (Razzeee) [inline xbmc/music/tags/MusicInfoTag.cpp]: We should probably use m_musicItemSeparator from advancedsettings here?
Not really part of this PR just came to mind.

Comment (Montellese) [inline xbmc/music/tags/MusicInfoTag.cpp]: You're right, I'm too much used to do this for JSON-RPC where we always want to use the same separator. `CAdvancedSettings::m_musicItemSeparator` and `CAdvancedSettings::m_videoItemSeparator` should theoretically only be used for concatenations that are shown in the GUI where it can't hurt anything. But you know our code base... ;-)

Comment (Razzeee) [inline xbmc/music/tags/MusicInfoTag.cpp]: Got it thanks, I think I've seen it more times then I like in some database files

**Comment by Montellese** [comment]
@Razzeee: Thanks for your feedback. I adjusted the string concatenation. jenkins build this please
